services/121-service/src/payments/services/payments-creation.service.ts

```diff
@@ -274,4 +306,106 @@ export class PaymentsCreationService {
       programFspConfigurationId: row.programFspConfigurationId,
     }));
   }
+
+  public async approvePayment({
+    userId,
+    programId,
+    paymentId,
+  }: {
+    userId: number;
+    programId: number;
+    paymentId: number;
+  }): Promise<void> {
```

Comment:
paymentCreationService.approvePayment is it a bit weird

1. maybe rename paymentCreationService to paymentManagementService?

2. Or create a paymentApprovmentService, as I don't think any existing functions are called in your new approval related functions?